components/authentication/authentication.js

```diff
@@ -134,6 +134,34 @@ function postRefresh(req,res){
     // A valid token for a deleted user profile can lead to
     // undefined behavior
     // should we make a DB query to see if the user is still active?
+    try{
+        let oldToken = {...req.auth}
+        // we don't need to check if it's null or undefined because the
+        // protectedRoute middleware did this before we got here.
+        if(oldToken.expired)
+            throw new Error("Expired token.")

```

Comment:
Instead of throwing an error, is there an HTTP response we can send instead? Throwing errors in the try - catch here will always return a `500` response.

Reply:
Done